lib/rewards/server/models/reward.js

```diff
@@ -0,0 +1,161 @@
+        this.wasGiven = param.wasGiven === 1;
+        this.hasDropBody = param.hasDropBody === 1;
+        this.animationData = param.animationData;
+    }
+
+    validate()
+    {
+        //TODO: Validate this object.
+    }
+
+    isWinningReward()
+    {
+        if (0 === this.dropRate) {
+            return false;
+        }
+        return Math.floor(Math.random() * 100) <= this.dropRate;
+    }
+
+
+    isValidReward()
+    {
+        if (0 === this.dropQuantity) {
+            return false;
+        }
+
+        if (this.isUnique && this.wasGiven) {
+            return false;
+        }
+
+        if (!this.isItemType() && !this.isModifierType() && !this.hasExperienceSet()) {
+            return false;
+        }
+
+        return true;
+    }
+
+    isItemType()
+    {
+        return null !== this.itemId;
+    }
+
+    isModifierType()
+    {
+        return null !== this.modifierId;
+    }
+
+    hasExperienceSet()
+    {
+        return 0 <= this.experience;
+    }
+
+    hasAnimationData()
+    {
+        return this.hasDropBody && sc.get(this, 'animationData', false);
+    }
+
+    static areValidRewards(rewards)
+    {
+        return 0 < (rewards?.length || 0);
+    }
+
+    static getRewardsBag(rewards)
+    {
+        let itemRangeArray = [];
+
+        for (let reward of rewards) {
+            let itemRangeCount = itemRangeArray.length;
+            for (let i = 0; i < reward.dropRate; i++) {
+                itemRangeArray[itemRangeCount + i] = reward;
+            }
+        }
+
+        return itemRangeArray;
+    }
+
+    static fromModel(rewardModel)
+    {
+        if (!rewardModel) {
+            return null;
+        }
+        return new Reward({
+            'id': rewardModel.id,
+            'objectId': rewardModel.object_id,
+            'itemId': rewardModel.item_id,
+            'modifierId': rewardModel.modifier_id,
+            'experience': rewardModel.experience,
+            'dropRate': rewardModel.drop_rate,
+            'dropQuantity': rewardModel.drop_quantity,
+            'isUnique': rewardModel.is_unique,
+            'wasGiven': rewardModel.was_given,
+            'hasDropBody': rewardModel.has_drop_body,
+            'animationData': ObjectsItemsRewardsAnimations.fromModel(rewardModel.objects_items_rewards_animations)
+        });
+    }
+
+    static createDropObjectData(reward, roomId)
+    {
+        return {
+            id: reward.randomRewardId+reward.tileIndex,
+            room_id: roomId,
+            layer_name: reward.randomRewardId,
+            tile_index: reward.tileIndex,
+            object_class_key: ObjectsConst.TYPE_DROP,
+            client_key: reward.randomRewardId+reward.tileIndex,
+            rewardModel: reward,
+            asset_key: reward.animationData.assetKey,
+            client_params: JSON.stringify({
+                frameStart: reward.animationData.extraParams.start,
+                frameEnd: reward.animationData.extraParams.end,
+                repeat: -1,
```

Comment:
this `repeat: -1` should be coming from reward.animationData.extraParams.repeat
and it could have a default config for the -1 if is not set

Reply:
OK!